### DISCUSSION TITLE
Symbol.for accepting string

### DISCUSSION THREAD

**Discussion by fedeci** [nodejs/help#4945] [Q&A] [answered]
Hi folks! According to the [ECMA specification](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-symbol.for) `Symbol.for()` should only accept a string as a key. However when running the following in the node shell it correctly creates a Symbol and it works nice for an array key. Am I missing something about the stability for non-string keys?

```js
Welcome to Node.js v18.3.0.
Type ".help" for more information.
> function a() { console.log("a") }
undefined
> const b = Symbol.for([a, 'foo'])
undefined
> const c = Symbol.for([a, 'foo'])
undefined
> b === c
true
> 
```

**Comment by Trott** [comment, marked as answer]
The first step in the algorithm in the spec section you link to is to convert `key` to a string value. 

> 1. Let stringKey be ? [ToString](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tostring)(key).

`Symbol.for()` is implemented in V8, not in Node.js code. So, if there was a bug, it needs to be opened against V8. 

That said, this does not appear to be a bug. The Firefox REPL exhibits the same behavior. So does the Chrome REPL. And the spec seems to indicate that `key` should be converted to a string.
